dcap/types/src/quote3.rs

```diff
@@ -0,0 +1,467 @@
+        no_qve_identity_data = { quote3_error_t::SGX_QL_NO_QVE_IDENTITY_DATA, Error::NoQveIdentityData },
+        unsupported_crl_format = { quote3_error_t::SGX_QL_CRL_UNSUPPORTED_FORMAT, Error::UnsupportedCrlFormat },
+        qe_identity_chain_error = { quote3_error_t::SGX_QL_QEIDENTITY_CHAIN_ERROR, Error::QeIdentityChainError },
+        tcb_info_chain_error = { quote3_error_t::SGX_QL_TCBINFO_CHAIN_ERROR, Error::TcbInfoChainError },
+        qvl_qve_mismatch = { quote3_error_t::SGX_QL_ERROR_QVL_QVE_MISMATCH, Error::QvlQveMismatch },
+        tcb_sw_hardening_needed = { quote3_error_t::SGX_QL_TCB_SW_HARDENING_NEEDED, Error::TcbSwHardeningNeeded },
+        tcb_configuration_and_sw_hardening_needed = { quote3_error_t::SGX_QL_TCB_CONFIGURATION_AND_SW_HARDENING_NEEDED, Error::TcbConfigurationAndSwHardeningNeeded },
+        unsupported_mode = { quote3_error_t::SGX_QL_UNSUPPORTED_MODE, Error::UnsupportedMode },
+        no_device = { quote3_error_t::SGX_QL_NO_DEVICE, Error::NoDevice },
+        service_unavailable = { quote3_error_t::SGX_QL_SERVICE_UNAVAILABLE, Error::ServiceUnavailable },
+        network_failure = { quote3_error_t::SGX_QL_NETWORK_FAILURE, Error::NetworkFailure },
+        service_timeout = { quote3_error_t::SGX_QL_SERVICE_TIMEOUT, Error::ServiceTimeout },
+        busy = { quote3_error_t::SGX_QL_ERROR_BUSY, Error::Busy },
+        unknown_message_response = { quote3_error_t::SGX_QL_UNKNOWN_MESSAGE_RESPONSE, Error::UnknownMessageResponse },
+        persistent_storage = { quote3_error_t::SGX_QL_PERSISTENT_STORAGE_ERROR, Error::PersistentStorage },
+        message_parsing = { quote3_error_t::SGX_QL_ERROR_MESSAGE_PARSING_ERROR, Error::MessageParsing },
+        platform_unknown = { quote3_error_t::SGX_QL_PLATFORM_UNKNOWN, Error::PlatformUnknown },
+        unknown_api_version = { quote3_error_t::SGX_QL_UNKNOWN_API_VERSION, Error::UnknownApiVersion },
+        certs_unavailable = { quote3_error_t::SGX_QL_CERTS_UNAVAILABLE, Error::CertsUnavailable },
+        qve_identity_mismatch = { quote3_error_t::SGX_QL_QVEIDENTITY_MISMATCH, Error::QveIdentityMismatch },
+        qve_out_of_date = { quote3_error_t::SGX_QL_QVE_OUT_OF_DATE, Error::QveOutOfDate },
+        psw_not_available = { quote3_error_t::SGX_QL_PSW_NOT_AVAILABLE, Error::PswNotAvailable },
+        collateral_version_not_supported = { quote3_error_t::SGX_QL_COLLATERAL_VERSION_NOT_SUPPORTED, Error::CollateralVersionNotSupported },
+        tdx_module_mismatch = { quote3_error_t::SGX_QL_TDX_MODULE_MISMATCH, Error::TdxModuleMismatch },
+        max = { quote3_error_t::SGX_QL_ERROR_MAX, Error::Max }
+    )]
+    fn error_from_ffi(ffi: quote3_error_t, expected: Error) {
+        assert_eq!(
+            expected,
+            Error::try_from(ffi).expect("Could not create error from ffi type")
+        )
+    }
+
+    #[test]
+    fn success_is_not_an_error() {
+        assert!(Error::try_from(quote3_error_t::SGX_QL_SUCCESS).is_err())
+    }
```

Comment:
:wrench: probably my coverage ocd
```suggestion
    }
    
    #[test]
    fn unknown_quote3_error_maps_to_unexpected() {
        let unknown = quote3_error_t(quote3_error_t::SGX_QL_ERROR_MAX.0 + 1);
        assert_eq!(
            Error::try_from(unknown).expect("Could not parse an unknown SGX Status"),
            Error::Unexpected
        );
    }
```